File: backend/app/controllers/user_controller.py
from flask import Blueprint, jsonify, request
from services.user_service import fetch_users_data

user_bp = Blueprint("user", __name__)

@user_bp.route("/users", methods=["GET"])
def get_users():
    user_id = request.cookies.get("user_id")
    if not user_id:
        return jsonify({"error": "未授權"}), 401

    user = fetch_users_data(user_id)
    if user and "error" in user:
        return jsonify(user), 404

    return jsonify(user)

File: backend/nest-app/src/controllers/user.controller.ts
import {
    Controller,
    Get,
    Req,
    Res,
    UseGuards,
    UnauthorizedException,
  } from '@nestjs/common';
  import { UserService } from '../services/user.service';
  import { Response, Request } from 'express';

  @Controller('user')
  export class UserController {
    constructor(private readonly userService: UserService) {}

    @Get('users')
    async getUsers(@Req() req: Request, @Res() res: Response) {
      const userId = req['user_id'];
      if (!userId) {
        throw new UnauthorizedException('未授權');
      }

      const result = await this.userService.fetchUsersData(userId);

      if ('error' in result) {
        return res.status(404).json(result);
      }

      return res.json(result);
    }
  }
